kubernetes/csi/lib/ubi_csi/node_service.rb

```diff
@@ -158,18 +167,26 @@ def perform_node_stage_volume(req_id, pvc, req, _call)
             log_with_id(req_id, "Loop device already exists: #{loop_device}")
           end
 
-          should_mkfs = is_new_loop_device
-          # in the case of copied PVCs, the previous has run the mkfs and by doing it again,
-          # we would wipe data so we avoid it here
-          if is_copied_pvc?(pvc)
-            should_mkfs = false
-          end
-          if !req.volume_capability.mount.nil? && should_mkfs
-            fs_type = req.volume_capability.mount.fs_type || "ext4"
-            output, ok = run_cmd("mkfs.#{fs_type}", loop_device, req_id:)
-            unless ok
-              log_with_id(req_id, "gRPC error in node_stage_volume: failed to format device: #{output}")
-              raise GRPC::Internal, "Failed to format device #{loop_device} with #{fs_type}: #{output}"
+          if req.volume_capability.mount
+            current_fs_type = find_file_system(loop_device, req_id:)
+            if !current_fs_type.empty? && !ACCEPTABLE_FS.include?(current_fs_type)
+              error_message = "Unacceptable file system type for #{loop_device}: #{current_fs_type}"
+              log_with_id(req_id, error_message)
+              raise error_message
```

Comment:
Consider adding a method that calls these two methods, and using that to DRY things up, since this pattern shows up in multiple places.